axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/projecttask/ProjectTaskGroupBusinessServiceImpl.java

```diff
@@ -117,4 +124,32 @@ public Map<String, Object> updateFinancialDatasWithoutPriceUnits(ProjectTask pro
     valuesMap.put("totalCosts", projectTask.getTotalCosts());
     return valuesMap;
   }
+
+  @Override
+  public Map<String, Object> updateInvoicingUnit(ProjectTask projectTask) throws AxelorException {
+    Map<String, Object> valuesMap = new HashMap<>();
+    projectTask.setQuantity(computeQuantity(projectTask));
+    projectTaskBusinessProjectService.updateDiscount(projectTask);
+    projectTaskBusinessProjectService.compute(projectTask);
+    valuesMap.put("quantity", projectTask.getQuantity());
+    valuesMap.put("unitPrice", projectTask.getUnitPrice());
+    valuesMap.put("unitCost", projectTask.getUnitCost());
+    valuesMap.put("exTaxTotal", projectTask.getExTaxTotal());
+    valuesMap.put("totalCosts", projectTask.getTotalCosts());
+    return valuesMap;
+  }
+
+  public BigDecimal computeQuantity(ProjectTask projectTask) throws AxelorException {
```

Comment:
In computeQuantity, we need to update unitPrice and unitCost with the reverse of the convert.

Ex : 
Days : 1 Qty and 700 unit price
Hours : 7 Qty and 100 unit price